File: 0547-number-of-provinces/0547-number-of-provinces.cpp
class Solution {
public:
    int findCircleNum(vector<vector<int>>& isConnected) {
             int n = isConnected.size();
        vector<int>adj[n];
        for(int i = 0;i<n;i++){
            for(int j = 0;j<n;j++){
                if(isConnected[i][j]){
                adj[i].push_back(j);
                adj[j].push_back(i);
                }
            }
        }
        vector<bool>vis(n,0);
        int ans= 0;
        for(int i = 0;i<n;i++){
            if(!vis[i]){
                ans++;
                queue<int>q;
                q.push(i);
                while(!q.empty()){
                    int node = q.front();
                    q.pop();
                    vis[node] = 1;
                    for(auto child:adj[node]){
                        if(!vis[child]){
                            q.push(child);
                        }
                    }
                }
            }
        }
        return ans;
    }
};

File: 0547-number-of-provinces/0547-number-of-provinces.py
class Solution:
    def findCircleNum(self, isConnected: List[List[int]]) -> int:
        row=len(isConnected)
        col=len(isConnected[0])
        visited=set()
        graph=defaultdict(list)
        self.ans=0
        def dfs(li):
            for l in li:
                if l not in visited:
                    visited.add(l)
                    dfs(graph[l])

        for i in range(row):
            for j in range(col):
                if isConnected[i][j]==1:
                    graph[i].append(j)
                    
        for g in graph:
            if g not in visited:
                self.ans+=1
                dfs(graph[g])
        return self.ans


